pennylane/shadows/classical_shadow.py

```diff
@@ -0,0 +1,370 @@
+            This method requires exponential scaling of measurements for accurate representations. Further, the output scales exponentially in the output dimension,
+            and is therefore not practical for larger systems. A warning is raised for systems of sizes ``n>16``.
+
+        Args:
+            wires (Iterable[int]): The wires over which to compute the snapshots. For ``wires=None`` (default) all ``n`` qubits are used.
+            snapshots (Iterable[int] or int): Only compute a subset of local snapshots. For ``snapshots=None`` (default), all local snapshots are taken.
+                In case of an integer, a random subset of that size is taken. The subset can also be explicitly fixed by passing an Iterable with the corresponding indices.
+
+        Returns:
+            tensor: The global snapshots tensor of shape ``(T, 2**n, 2**n)`` containing the density matrices for each snapshot measurement.
+
+        **Example**
+
+        We can approximately reconstruct a Bell state:
+
+        .. code-block:: python3
+
+            dev = qml.device("default.qubit", wires=range(2), shots=1000)
+            @qml.qnode(dev)
+            def qnode():
+                qml.Hadamard(0)
+                qml.CNOT((0,1))
+                return classical_shadow(wires=range(2))
+
+            bits, recipes = qnode()
+            shadow = ClassicalShadow(bits, recipes)
+            shadow_state = np.mean(shadow.global_snapshots(), axis=0)
+
+            bell_state = np.array([[0.5, 0, 0, 0.5], [0, 0, 0, 0], [0, 0, 0, 0], [0.5, 0, 0, 0.5]])
+
+            >>> np.allclose(bell_state, shadow_state, atol=1e-1)
+            True
+
+        """
+
+        local_snapshot = self.local_snapshots(wires, snapshots)
+
+        if local_snapshot.shape[1] > 16:
+            warnings.warn(
+                "Querying density matrices for n_wires > 16 is not recommended, operation will take a long time",
+                UserWarning,
+            )
+
+        return self._obtain_global_snapshots(local_snapshot)
+
+    def _convert_to_pauli_words(self, observable):
+        """Given an observable, obtain a list of coefficients and Pauli words, the
+        sum of which is equal to the observable"""
+
+        num_wires = self.bits.shape[1]
+        obs_to_recipe_map = {"PauliX": 0, "PauliY": 1, "PauliZ": 2, "Identity": -1}
+
+        def pauli_list_to_word(obs):
+            word = [-1] * num_wires
+            for ob in obs:
+                if ob.name not in obs_to_recipe_map:
+                    raise ValueError("Observable must be a linear combination of Pauli observables")
+
+                word[ob.wires[0]] = obs_to_recipe_map[ob.name]
+
+            return word
+
+        if isinstance(observable, (qml.PauliX, qml.PauliY, qml.PauliZ, qml.Identity)):
+            word = pauli_list_to_word([observable])
+            return [(1, word)]
+
+        if isinstance(observable, qml.operation.Tensor):
```

Comment:
```suggestion
        elif isinstance(observable, qml.operation.Tensor):
```

Reply:
I had this before, but pylint complains about the unecessary elif after a return statement.

Reply:
Yep, you are right! Didn't see the return statement.